### PR TITLE
[#1] Fulfil the open-source check-list: implemented example.

### PR DESCRIPTION
**Pull request by cabol**
[#1] Fulfil the open-source check-list: implemented example.

### CODE REVIEW

File: example/priv/index.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<h2>This is a Cowboy Trails example!</h2>
+<h3><a href="http://localhost:8080/description">Cowboy Trails Descriptions</a></h3>
+<br>
+<a href="https://en.wikipedia.org/wiki/Cowboy_Trail">The Cowboy Trail is a rail trail in northern Nebraska</a>
```
Comment (elbrujohalcon) [inline example/priv/index.html]: This is too related to trails… make it _swagger-y_

Comment (elbrujohalcon) [inline example/priv/index.html]: The whole page, please

File: example/src/example_echo_handler.erl [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            required => false,
+            type => <<"string">>}
+        ]
+      }
+    },
+  [trails:trail("/message/[:echo]", example_echo_handler, [], Metadata)].
+
+%% cowboy
+allowed_methods(Req, State) ->
+  {[<<"GET">>, <<"PUT">>,   <<"HEAD">>], Req, State}.
```
Comment (elbrujohalcon) [inline example/src/example_echo_handler.erl]: too many spaces

File: example/src/example_echo_handler.erl [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+%% internal
+handle_get(Req, State) ->
+  Echo = application:get_env(example, echo, ""),
+  Body = [<<"You Get an echo! ">> , Echo],
```
Comment (HernanRivasAcosta) [inline example/src/example_echo_handler.erl]: ``Body = [<<"You got an echo!">> , Echo], ``

**Pull request merged**